src/main/java/com/google/cloud/tools/gradle/appengine/appyaml/AppEngineAppYamlPlugin.java

```diff
@@ -107,44 +109,43 @@ private void configureExtensions() {
             deploy.setAppEngineDirectory(stageExtension.getAppEngineDirectory());
           }
 
-          DeployAllTask deployAllTask =
-              (DeployAllTask)
-                  project
-                      .getTasks()
-                      .getByName(AppEngineCorePluginConfiguration.DEPLOY_ALL_TASK_NAME);
-          deployAllTask.setStageDirectory(stageExtension.getStagingDirectory());
-          deployAllTask.setDeployExtension(deploy);
-
-          DeployTask deployTask =
-              (DeployTask)
-                  project.getTasks().getByName(AppEngineCorePluginConfiguration.DEPLOY_TASK_NAME);
-          deployTask.setDeployConfig(deploy);
-          deployTask.setAppYaml(stageExtension.getStagingDirectory().toPath().resolve("app.yaml"));
+          tasks
+              .withType(DeployAllTask.class)
+              .configureEach(
+                  task -> {
+                    task.setStageDirectory(stageExtension.getStagingDirectory());
+                    task.setDeployExtension(deploy);
```

Comment:
If you use `Property<...>` for `stageExtension.getStagingDirectory`, then you could move this task configuration outside of `afterEvaluate`.

Reply:
Sounds like a great suggestion, but unfortunately, I don't really follow what you are talking about. (I don't have much experience with Gradle.) And perhaps this kind of refactoring can be done incrementally in a separate PR. Maybe you could file an issue with an explanation of a potential solution?